cmdstanpy/utils.py

```diff
@@ -629,6 +629,31 @@ def scan_column_names(fd: TextIO, config_dict: Dict, lineno: int) -> int:
     return lineno
 
 
+def parse_var_dims(names: Tuple[str, ...]) -> Dict:
+    """
+    Use Stan CSV file column names to get variable names, dimensions.
+    Assumes that CSV file has been validated and column names are correct.
+    """
+    if names is None:
+        raise ValueError('missing argument "names"')
+    vars_dict = {}
+    idx = 0
+    while idx < len(names):
+        if names[idx].endswith('__'):
+            pass
+        elif '.' not in names[idx]:
+            vars_dict[names[idx]] = 1
+        else:
+            vs = names[idx].split('.')
+            if idx < len(names) - 1 and names[idx + 1].split('.')[0] == vs[0]:
+                idx += 1
+                continue
+            dims = [int(vs[x]) for x in range(1, len(vs))]
```

Comment:
Does this work for ragged structures?

Reply:
no, this won't work for ragged structures - they'd need to be coded differently.
does numpy allow ragged ndarrays?

Reply:
It kind of does, but really it doesn't

numpy mask might help.

But having non-ragged structure with nan might work

Reply:
there is a proposal for adding ragged arrays to the language - also tuples will be coming soon - so I think the next step will be to re-implement using xarray.